boa/src/syntax/parser/class/mod.rs

```diff
@@ -0,0 +1,334 @@
+//! Function definition parsing.
+//!
+//! More information:
+//!  - [MDN documentation][mdn]
+//!  - [ECMAScript specification][spec]
+//!
+//! [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/function
+//! [spec]: https://tc39.es/ecma262/#sec-function-definitions
+
+#[cfg(test)]
+mod tests;
+
+use crate::{
+    gc::{Finalize, Trace},
+    syntax::{
+        ast::{
+            node::{FunctionDecl, Node},
+            Keyword, Punctuator,
+        },
+        lexer::{InputElement, TokenKind},
+        parser::{
+            expression::Expression,
+            function::{FormalParameters, FunctionBody},
+            statement::BindingIdentifier,
+            AllowAwait, AllowYield, Cursor, ParseError, TokenParser,
+        },
+    },
+    BoaProfiler,
+};
+use std::{
+    collections::{HashMap, HashSet},
+    io::Read,
+};
+
+#[cfg(feature = "deser")]
+use serde::{Deserialize, Serialize};
+
+#[cfg_attr(feature = "deser", derive(Serialize, Deserialize))]
+#[derive(Clone, Debug, Trace, Finalize, PartialEq)]
+pub enum ClassField {
+    /// A method on a class.
+    Method(FunctionDecl),
+    /// A field on a class (includes an initializer)
+    // TODO: Name should be a VariableDeclList (I think)
+    Field(Box<str>, Node),
+    /// A getter function. This will never take any arguments.
+    Getter(FunctionDecl),
+    /// A setter function. This will always take an argument.
+    Setter(FunctionDecl),
+}
```

Comment:
This should live with the node, not with the parser, I think. Also, it requires some documentation.